src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -196,6 +196,30 @@ public function findByPidFull(string $pid, string $entityType = 'CoreEntity'): ?
         return $this->resolveCategories($withHydratedParents)[0];
     }
 
+    /**
+     * @param string[] $pids
+     * @param string $entityType
+     * @return array
+     */
+    public function findByPids(array $pids, string $entityType = 'CoreEntity'): array
```

Comment:
What if you pass in 4 pids, but only 3 are found. Do we want to error or anything?

Reply:
No, I think that's legitmate behaviour that the calling function should handle.

Reply:
I think it's fine if we're consistent with the behaviour of findByPid and just return what is found & return an empty array if nothing.

Reply:
An example of this occuring is if a programme is embargoed. In this case we'd want to return the programmes without the embargoed item rather than error. Adding an embargoed item somewhere may be an editorial decision so it appears as soon as the embargo is lifted. No need to punish the user for that editorial workflow.